### PR TITLE
procmail: 3.22 (new formula)

### PR DESCRIPTION
**Pull request by knakayama**
Looks like procmail is being removed in [El Capitan.](http://www.shirt-pocket.com/blog/index.php/shadedgrey/comments/oh_yeah_well_lock_this/) And there is some demand for this PR(#43679), so I re-submit PR.

### CODE REVIEW

File: Library/Formula/procmail.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+class Procmail < Formula
+  desc "procmail: sort incoming mail into separate folders/files"
+  homepage "http://www.procmail.org/"
+  url "ftp://ftp.procmail.net/pub/procmail/procmail-3.22.tar.gz"
+  mirror "ftp://ftp.ucsb.edu/pub/mirrors/procmail/procmail-3.22.tar.gz"
+  sha256 "087c75b34dd33d8b9df5afe9e42801c9395f4bf373a784d9bc97153b0062e117"
+
+  depends_on "autoconf" => :build
+
+  patch :p1, :DATA
```
Comment (bfontaine) [inline Library/Formula/procmail.rb]: Please add a comment explaining why this patch is necessary as well as a link to the upstream submission. Thanks!

**Comment by DomT4** [comment]
@Homebrew/owners Is it worth creating a `keg_only :provided_pre_el_capitan` here?

It's not any newer or older than the system version, there hasn't been a release in 14 years, but don't know if we want to prevent it overriding the system version on <10.11 regardless.

**Comment by mikemcquaid** [comment]
> @Homebrew/owners Is it worth creating a keg_only :provided_pre_el_capitan here?

:+1:

**Comment by DomT4** [comment]
https://github.com/Homebrew/homebrew/commit/ad1fcfc77c0871fe02366687be32c766bf89df20

@knakayama If you rebase your PR against the master branch here you should be able to use:

```ruby
keg_only :provided_pre_el_capitan
```

in the formula, one empty line underneath the `depends_on` line. Thanks!

**Comment by knakayama** [comment]
I add comments about patch add `keg_only` line. My English skill is terrible, so it may be a little strange...

> as well as a link to the upstream submission

@bfontaine 
It looks like procmail has not been [developed](http://www.procmail.org/procmail.HISTORY.html) (3.22 is released in 2001). So I have not submitted patch because it will not be merged and Macports procmail has [the same approach](https://github.com/dstrctrng/macports/blob/master/rsync.macports.org/release/tarballs/ports/mail/procmail/files/getline.patch).

**Comment by mikemcquaid** [comment]
> It looks like procmail has not been developed (3.22 is released in 2001). So I have not submitted patch because it will not be merged and Macports procmail has the same approach.

We are different to MacPorts and require all patches to be submitted upstream.

**Comment by knakayama** [comment]
> We are different to MacPorts and require all patches to be submitted upstream.

OK, I appreciate the homebrew way. But I have no time to submit patches for now.
Is there anyone who submit patches? If no one, I don't mind that this PR is rejected. Thanks.

**Comment by bmike** [comment]
I'd like to help move this forward. What's the path forward?

**Comment by bfontaine** [comment]
@bmike we need to submit the patch to upstream.

File: Library/Formula/procmail.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
+class Procmail < Formula
+  desc "procmail: sort incoming mail into separate folders/files"
```
Comment (DomT4) [inline Library/Formula/procmail.rb]: You don't need the formula name in here. It'll be prepended automatically.

```ruby
desc "Sort incoming mail into separate folders/files"
```

is fine.

**Comment by knakayama** [comment]
Thanks. I modify the description.

**Comment by afh** [comment]
How about using the sources that Apple provides at [http://opensource.apple.com](http://www.opensource.apple.com/tarballs/procmail) and that were used for previous versions of Mac OS X (see #44105)?

**Comment by knakayama** [comment]
> How about using the sources that Apple provides at http://opensource.apple.com and that were used for previous versions of Mac OS X (see #44105)?

@Homebrew/owners
If use this source, does not need **upstream submission**?

**Comment by mikemcquaid** [comment]
> If use this source, does not need upstream submission?

If it's Apple's: nope.

**Comment by afh** [comment]
I'd like to mention that there's a procmail formula pending in #44105. It has functional tests and compiles to `brew audit --strict`. An issue one might have with it is that is uses a different (Apple's) version number (14) than the procmail upstream (3.22). Just sayin'… :wink:

**Pull request closed without merging**